File: LeetCode/LC.2380.time-needed-to-rearrange-a-binary-string.cpp
#include "../utils/abel_macro.h"

class Solution {
public:
    int secondsToRemoveOccurrences(string s) {
        int f = 0, pre = 0;
        for (char c : s) {
            if (c == '0') {
                pre++;
            } else if (pre) {
                f = max(f + 1, pre);
            }
        }
        return f;
    }
};

File: LeetCode/LC.2380.time-needed-to-rearrange-a-binary-string.py
# key: pre0

class Solution:
    def secondsToRemoveOccurrences(self, s: str) -> int:
        f = pre0 = 0
        for c in s:
            if c == '0': pre0 += 1
            elif pre0: f = max(f + 1, pre0)
        return f